main.nf

```diff
@@ -1,277 +1,985 @@
 }
 log.info summary.collect { k,v -> "${k.padRight(18)}: $v" }.join("\n")
-log.info "\033[2m----------------------------------------------------\033[0m"
+log.info "-\033[2m--------------------------------------------------\033[0m-"
 
 // Check the hostnames against configured profiles
 checkHostname()
 
-def create_workflow_summary(summary) {
-    def yaml_file = workDir.resolve('workflow_summary_mqc.yaml')
-    yaml_file.text  = """
+Channel.from(summary.collect{ [it.key, it.value] })
+    .map { k,v -> "<dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }
+    .reduce { a, b -> return [a, b].join("\n            ") }
+    .map { x -> """
     id: 'nf-core-cageseq-summary'
     description: " - this information is collected when the pipeline is started."
     section_name: 'nf-core/cageseq Workflow Summary'
     section_href: 'https://github.com/nf-core/cageseq'
     plot_type: 'html'
     data: |
         <dl class=\"dl-horizontal\">
-${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }.join("\n")}
+            $x
         </dl>
-    """.stripIndent()
-
-   return yaml_file
-}
+    """.stripIndent() }
+    .set { ch_workflow_summary }
 
-
-/*
- * Parse software version numbers
- */
 process get_software_versions {
+    publishDir "${params.outdir}/pipeline_info", mode: params.publish_dir_mode,
+        saveAs: { filename ->
+                    if (filename.indexOf(".csv") > 0) filename
+                    else null
+        }
 
     output:
-    file 'software_versions_mqc.yaml' into software_versions_yaml
+    file 'software_versions_mqc.yaml' into ch_software_versions_yaml
+    file "software_versions.csv"
 
     script:
-    // TODO nf-core: Get all tools to print their version number here
     """
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
     fastqc --version > v_fastqc.txt
     multiqc --version > v_multiqc.txt
-    scrape_software_versions.py > software_versions_mqc.yaml
+    STAR --version > v_star.txt
+    bowtie --version > v_bowtie.txt
+    cutadapt --version > v_cutadapt.txt
+    samtools --version > v_samtools.txt
+    bedtools --version > v_bedtools.txt
+    read_distribution.py --version > v_rseqc.txt
+    sortmerna --version > v_sortmerna.txt
+    scrape_software_versions.py &> software_versions_mqc.yaml
     """
 }
+if(!params.skip_ctss_qc){
+    process convert_gtf {
+        tag "$gtf"
 
+        input:
+        file gtf from gtf_rseqc
 
+        output:
+        file "${gtf.baseName}.bed" into bed_rseqc
 
+        script:
+        """
+        gtf2bed.pl $gtf > ${gtf.baseName}.bed
+        """
+    }
+}
+if(!params.skip_ctss_generation){
+    process get_chrom_sizes{
+    input:
+    file fasta from fasta_rseqc.collect()
+    output:
+    file "*.txt" into (chrom_sizes_ctss,chrom_sizes_bw)
+
+    shell:
```

Comment:
Why are you using `shell` here just out of interest?

Reply:
I switched to `shell`, because the awk script didn't work in a `script` element (I think due to a conflict in the variable naming)

Reply:
I previously encountered the same issue and used the same solution, for reference: https://github.com/nextflow-io/nextflow/issues/157